### PR TITLE
[SPARK-3298] [SQL] registerAsTable/registerTempTable overwrites old tables 

### PR DESCRIPTION
**Pull request by animeshbaranawal**
@akhilthatipamula

Added exception handling to prevent overwriting an existing table with the same name.

The contribution is my original work and I license the work to the project under the project's open source license.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala [hunk 1]
```diff
@@ -1193,12 +1193,18 @@ class DataFrame private[sql](
...
+   * Check if the table name already exists
+   * Registers this [[DataFrame]] as a temporary table using the given name if 
+   * no such table name present else throw an exception 
+   * The lifetime of this temporary table is tied to  
+   * the [[SQLContext]] that was used to create this DataFrame.
+   * 
    * @group basic
    */
   def registerTempTable(tableName: String): Unit = {
+    if( sqlContext.catalog.tableExists(Seq(tableName)) ){
```
Comment (chenghao-intel) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala]: Nit: space after `if`, and no space after `(` or before `)`.
Says:
```scala
if (sqlContext.catalog.tableExists(Seq(tableName))) {
```

**Comment by chenghao-intel** [comment]
As `registerTempTable` probably invoked in multiple threads, I am not sure if here is the best place to check the table existence, probably move the logic inside of `catalog`?

**Comment by animeshbaranawal** [comment]
So should I apply the same logic to registerDataframeAsTable in the SQLContext.scala file ?

**Comment by marmbrus** [comment]
Thanks for working on this, but as discussed on the JIRA we can't actually make this change without breaking existing user code.  As a result, I think we are stuck with the existing behavior and should close this issue unless there is a really compelling reason for adding the complexity of a configuration flag.

**Pull request closed without merging**